magni_description/urdf/sensors/sonar_hc-sr04.xacro

```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
+<robot xmlns:xacro="http://wiki.ros.org/xacro">
+
+  <xacro:macro name="sonar_hc-sr04" params="name:='' connected_to:='' *origin">
+
+    <xacro:unless value="${connected_to == ''}">
+      <joint name="${name}_joint" type="fixed">
+        <parent link="${connected_to}"/>
+        <child link="${name}"/>
+        <xacro:insert_block name="origin"/>
+      </joint>
+    </xacro:unless>
+
+    <link name="${name}">
+      <visual>
+        <origin xyz="-0.005 0 0" rpy="${pi/2} 0 ${pi/2}" />
+        <geometry>
+          <mesh filename="package://magni_description/meshes/sensors/sonar_hc-sr04.dae" />
+        </geometry>
+      </visual>
+      <collision>
+        <origin xyz="0.0075 0 0" rpy="0 0 0" />
+        <geometry>
+          <box size="0.02 0.044 0.021"/>
+        </geometry>
+      </collision>
+    </link>
+
+
+    <!-- Gazebo sensor plugin -->
+    <gazebo reference="${name}">
+      <sensor type="ray" name="${name}_sensor">
+        <visualize>false</visualize>
+        <update_rate>5</update_rate>
+        <ray>
+          <scan>
+            <horizontal>
+              <samples>5</samples>
+              <resolution>1.0</resolution>
+              <min_angle>-0.24</min_angle>
+              <max_angle>0.24</max_angle>
+            </horizontal>
+            <vertical>
+              <samples>5</samples>
+              <resolution>1.0</resolution>
+              <min_angle>-0.24</min_angle>
+              <max_angle>0.24</max_angle>
+            </vertical>
+          </scan>
+          <range>
+            <min>0.02</min>
+            <max>3.2</max>
+            <resolution>0.01</resolution>
+          </range>
+        </ray>
+        <!-- Gazebo-ROS interface plugin -->
+        <plugin filename="libgazebo_ros_range.so" name="gazebo_ros_range">
+          <gaussianNoise>0.005</gaussianNoise>
```

Comment:
The noise on this sensor should be higher, probably 0.015 is more correct.

Reply:
I moved camera sensor forward a little to avoid occluding with sonar.
All other fixes are finished.
Please check updated PR.